consensus/enclave/impl/src/lib.rs

```diff
@@ -564,7 +599,7 @@ impl ConsensusEnclave for SgxConsensusEnclave {
 
         // Form the block.
         let block = Block::new_with_parent(
```

Comment:
Maybe this should actually be the place to check that `block_version >= parent_block.version`?
Not necessarily in this PR since it's an annoying change to have it return a `Result`.